pkg/verify/util/encode.go

```diff
@@ -0,0 +1,40 @@
+package util
+
+import (
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/serializer"
+)
+
+var (
+	coreScheme  = runtime.NewScheme()
+	coreCodecs  = serializer.NewCodecFactory(coreScheme)
+	coreEncoder runtime.Encoder
+)
+
+func init() {
+	if err := corev1.AddToScheme(coreScheme); err != nil {
+		panic(err)
+	}
+	coreEncoderCodecFactory := serializer.NewCodecFactory(coreScheme)
+	coreEncoder = coreEncoderCodecFactory.LegacyCodec(corev1.SchemeGroupVersion)
+}
+
+// ReadConfigMap reads config map object from bytes. nil is returned if
+// the object cannot be decoded as a config map.
+func ReadConfigMap(objBytes []byte) (*corev1.ConfigMap, error) {
```

Comment:
Seems like this function should go into a separate helper, outside of `verify`, and it can replace [our existing `ReadConfigMapV1OrDie`][1] (we'd need to `panic` locally if the library function returned an error).  And I guess move `ConfigMapAsBytes` over too?  But whatever, doesn't need to block the move to library-go.

[1]: https://github.com/openshift/cluster-version-operator/blob/a49fef5c66c6b0707c54fd93f84d2f51d3d28aca/lib/resourceread/core.go#L21-L26

Reply:
I thought of same but simply wanted to concentrate on what was immediately required for this reuse task.